packages/ses/src/tame-faux-data-properties.js

```diff
@@ -0,0 +1,124 @@
+  if (subject1[prop] !== testValue) {
+    return false;
+  }
+  const subject2 = { __proto__: obj };
+  apply(set, subject2, [testValue]);
+  if (subject2[prop] !== testValue) {
+    return false;
+  }
+
+  // TODO We would like to test that the setter throws when attempting to
+  // set the target object. However, since we don't already know what it
+  // does, the following test could cause a persistent side effect.
+  // So it is commented out for now.
+  // if (!throws(() => apply(set, obj, [testValue]))) {
+  //   return false;
+  // }
+
+  if ('originalValue' in get) {
+    return false;
+  }
+
+  // We assume that this code runs before any untrusted code runs, so
+  // we do not need to worry about the above conditions passing because of
+  // malicious intent. In fact, it runs even before vetted shims are supposed
+  // to run, between repair and hardening. Given that, after all these tests
+  // pass, we have adequately validated that the property in question is
+  // an accessor function whose purpose is suppressing the override mistake,
+  // i.e., enabling a non-writable property to be overridden by assignment.
+  // In that case, here we *temporarily* turn it into the data property
+  // it seems to emulate, but writable so that it does not trigger the
+  // override mistake while in this temporary state.
+
+  // For those properties that are also listed in `enablements.js`,
+  // that phase will re-enable override for these properties, but
+  // via accessor functions that ses controls, so we know what they are
+  // doing. In addition, the getter functions installed by
+  // `enable-property-overrides.js` have an `originalValue` field
+  // enabling meta-traversal code like harden to visit the original value
+  // without calling the getter.
+
+  if (desc.configurable === false) {
+    // Even though it seems to be an accessor, we're unable to fix it.
+    return false;
+  }
+
+  // Many of the `return false;` cases above plausibly should be turned into
+  // errors, or an least generate warnings. However, for those, the checks
+  // following this phase are likely to signal an error anyway.
+
+  defineProperty(obj, prop, {
+    value: expectedValue,
+    writable: true,
+    enumerable: desc.enumerable,
+    configurable: true,
+  });
+
+  return true;
+};
+
+/**
+ * For each, see it the property is an accessor property that
+ * seems to emulate a data property with this expectedValue,
+ * and seems to be an accessor whose purpose is to protect against
+ * the override mistake, i.e., enable these properties to be overridden
+ * by assignment. If all these expected conditions are met, then
+ * *temporarily* turn it into the data property it emulated.
+ *
+ * Coordinate with `enablements.js` so the appropriate ones are
+ * turned back to accessor that protect against override mistake,
+ * but accessors we know.
+ *
+ * @param {Record<any,any>} intrinsics
+ */
+export const tameFauxDataProperties = intrinsics => {
```

Comment:
What distinguishes an “enablement” from a “faux data property”? Are they the same?